### PR TITLE
Issue #13999: kill mutation for JavadocMethodCheck

### PR DESCRIPTION
**Pull request by mohitsatr**
part of #13999:

mutation in question:
https://github.com/checkstyle/checkstyle/blob/24eedce6a64145442db5b478c7a39ce0763dad8c/config/pitest-suppressions/pitest-javadoc-suppressions.xml#L57-L64

**Rationale:**
`surroundingAccessModifier ` is one of the deciding factor if  we want to process the method/construction or not. the case `surroundingAccessModifier == null` is when we come on a overriden method. For example : 
```
final Runnable r = new Runnable() {
    public void run() {};
};
```
this condition might have been placed there because: 
 - it's better not to send surroundingAccessModifier down in camparsions because with null there is always risk of NullPointerException error. 
 - this simple checking saves us making unnecessary camparisons.

**Removal is OK**  because surroundingAccessModifier is something we are only camparing with each modifier object. we are not calling a method call on surroundingAccessModifier.
in short `==` there is no risk of running into NullPointerException error here.

### CODE REVIEW

**Comment by romani** [comment]
Github, generate report for JavadocMethod/all-examples-in-one

**Comment by romani** [comment]
If diff report shows no difference, we can merge

**Comment by romani** [comment]
@mohitsatr , can you explain analytically why removal is ok?

in practice(diff report) we see that it can be removed.
Ths is not ast based Check so we are accepting some theoretically regression on code that is not part of our regression.

**Comment by mohitsatr** [comment]
> @mohitsatr , can you explain analytically why removal is ok?

done 
the PR was in draft because I was still figuring out how to generate diff report.

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/blob/master/.github/workflows/README.md#regression-report-workflow---regression-reportyml

Or

https://github.com/checkstyle/contribution/tree/master/checkstyle-tester#executing-generation-using-github-action

------

I already did at https://github.com/checkstyle/checkstyle/pull/16031#issuecomment-2541771918

**Comment by mohitsatr** [comment]
CI is read . is it me ? should I push again ?

**Review by romani — APPROVED** [review]
Ok to merge

**Review by nrmancuso — APPROVED** [review]
Thanks a lot @mohitsatr
